gen/fs_home

```diff
@@ -147,16 +153,117 @@ foreach my $login (sort keys %$memberAttributesByLoginAndMount) {
 
 close(SERVICE_FILE);
 
+#----------------------------------------------------------
+# QUOTAS DATA
+#----------------------------------------------------------
+
+#structured data about quotas
+my $dataForUserQuotasByUID = {};
+
+open QUOTAS_FILE,">$quotas_file_name" or die "Cannot open $quotas_file_name: $! \n";
+
+#Temporary define if facility is ready for new quotas (dataQuotas and fileQuotas)
+my $readyForNewQuotas = $facilityAttributes{$A_F_NEW_QUOTAS_READY};
+
+#prepare quotas data
+foreach my $rData (@resourcesData) {
+	my %resourceAttributes = attributesToHash $rData->getAttributes;
+	my @membersData = ($rData->getChildElements)[1]->getChildElements;
+
+	if($readyForNewQuotas) {
+		foreach my $mData (@membersData) {
+			my %memberAttributes = attributesToHash $mData->getAttributes;
+			my $uid = $memberAttributes{$A_UID};
+
+			unless($dataForUserQuotasByUID->{$uid}) {
+				#First go through data quota map and read file quota if exists or set unlimited
+				foreach my $volume (keys %{$memberAttributes{$A_UF_DATA_QUOTAS}}) {
+					my $dataQuota = $memberAttributes{$A_UF_DATA_QUOTAS}{$volume};
+
+					my $softDataQuota = $dataQuota;
+					$softDataQuota =~ s/:.*$//g;
+					my $hardDataQuota = $dataQuota;
+					$hardDataQuota =~ s/^.*://g;
+					$dataForUserQuotasByUID->{$uid}->{$volume}->{$dataQuotaHeader} = quotaToKb $softDataQuota;
+					$dataForUserQuotasByUID->{$uid}->{$volume}->{$dataLimitHeader} = quotaToKb $hardDataQuota;
+
+					if($memberAttributes{$A_UF_FILE_QUOTAS}{$volume}) {
+						my $fileQuota = $memberAttributes{$A_UF_FILE_QUOTAS}{$volume};
+						my $softFileQuota = $fileQuota;
+						$softFileQuota =~ s/:.*$//g;
+						my $hardFileQuota = $fileQuota;
+						$hardFileQuota =~ s/^.*://g;
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$fileQuotaHeader} = $softFileQuota;
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$fileLimitHeader} = $hardFileQuota;
+					} else {
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$fileQuotaHeader} = 0;
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$fileLimitHeader} = 0;
+					}
+				}
+
+				#Then go through file quota map and for not used files quota set unlimited data quota
+				foreach my $volume (keys %{$memberAttributes{$A_UF_FILE_QUOTAS}}) {
+					#if this path was not processed yet (not set for data quotas at all)
+					unless($dataForUserQuotasByUID->{$uid}->{$volume}) {
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$dataQuotaHeader} = 0;
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$dataLimitHeader} = 0;
+						my $fileQuota = $memberAttributes{$A_UF_FILE_QUOTAS}{$volume};
+						my $softFileQuota = $fileQuota;
+						$softFileQuota =~ s/:.*$//g;
+						my $hardFileQuota = $fileQuota;
+						$hardFileQuota =~ s/^.*://g;
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$fileQuotaHeader} = $softFileQuota;
+						$dataForUserQuotasByUID->{$uid}->{$volume}->{$fileLimitHeader} = $hardFileQuota;
+					}
+				}
```

Comment:
Lines 205-218 are just repetition of lines 190-197. It's very confusing that you are processing files quota on two places - first during processing of files quota and then once again to add files quota for volumes which were not processed before.
I'd suggest to write this differently.

Reply:
You are right. I just copied it from fs_quotas, but it is confusing. Now after rework it should be less confusing I hope.

Reply:
It's way better :+1: